assets/global.js

```diff
@@ -68,6 +68,13 @@ function trapFocus(container, elementToFocus = container) {
   document.addEventListener('focusin', trapFocusHandlers.focusin);
 
   elementToFocus.focus();
+
+  if (elementToFocus.tagName === 'INPUT' &&
+    ['search', 'text', 'email', 'url'].includes(elementToFocus.type) &&
+    elementToFocus.value) {
+    const end = elementToFocus.value.length;
+    elementToFocus.setSelectionRange(0, end);
+  }
```

Comment:
since this is in the `trapFocus()` function this will only run when we're trapping the focus. But won't happen anywhere else on inputs. So for example on the `about us` page for the `os2-demo` store, it won't do anything on the contact us form fields.

We currently run `trapFocus()` on: 
- the cart drawer
- the cart notification
- modals using the `detais-modal` custom HTML element (search, password page modal to enter password)
- the menu drawer 
- the `modal-dialog` custom HTML element
- the pickup availability drawer

In all those cases, I think the search is the only scenario where the element receiving focus due to our `trapFocus` function is an `input` element. 

I'm not sure if introducing this behaviour only for that one specific part of the theme makes sense when the rest of the input don't get the same behaviour on focus 🤔 

But then having a listener for `focusin` on all inputs of the theme might be a bit much. Not sure. 

What do you think ? 🤔

Reply:
>Adding this on every focusin event will select the value also when the user focuses the input by clicking on it, which I don't think is a good UX since they might be clicking exactly in the point were they want their cursor.
💯 

I'll approve the PR 👍 Though we should wait a bit before merging (once you get the second approval).